execution/gethexec/executionengine.go

```diff
@@ -624,6 +632,14 @@ func (s *ExecutionEngine) digestMessageWithBlockMutex(num arbutil.MessageIndex,
 	if err != nil {
 		return err
 	}
+
+	if s.consensus != nil {
```

Comment:
I don't like BroadcastMessage as part of the consensus interface.
That creates a weird processing order consensus->execution->consensus
I see what you gain from it in case of reorg, but I don't think it's worth it. A "BroadcastMEssage" API doesn't make too much sense in the grand scheme (e.g. if you think about one consensus <-> multiple executions)

Reply:
Now ExecutionEngine returns MessageResults to TransactionStreamer to avoid having a BroadcastMessage func in TransactionStreamer.